### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by tinohager**
This pull request includes significant updates to the `README.md` file to enhance clarity, improve structure.

Enhancements to the `README.md`:

* Rewritten introduction to provide a clearer and more engaging description of Lychee, including a mention of the Supporter Edition.
* Reorganized installation instructions into three clear deployment options: file-based, Docker, and build from source.
* Moved and updated the notice about the `master` and `alpha` branches to provide better context on the development process and encourage contributions.

### CODE REVIEW

**Review by ildyria — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -14,41 +14,36 @@
...
+### File-based deployment
 
-We aim to maintain a free open-source photography library with high quality of code.<br>
-Being in control of our own data, our own pictures is something that we value above all.
+Copy the extracted Zip file from https://github.com/LycheeOrg/Lychee/releases to your webserver
 
-Through [contributions, donations, and sponsorship](https://github.com/sponsors/LycheeOrg), you allow Lychee to thrive. Your donations directly support demo server costs, continuous enhancements, and most importantly bug fixes!
+### Docker deployment
 
+Alternatively, an official Docker image can be found at [LycheeOrg/Lychee-Docker](https://github.com/LycheeOrg/Lychee-Docker) or on Docker Hub as [lycheeorg/lychee](https://hub.docker.com/r/lycheeorg/lychee).
```
Comment (d7415) [inline README.md]: This should probably be above File-based if we're recommending it.

Comment (tinohager) [reply README.md]: File-based has the disadvantage that many dependencies such as ffmpeg or the php version must fit. This catalog of dependencies does not exist with docker and the installation of the lychee docker variant requires 3 commandos. So maybe the way should point in that direction it is definitely the way I would recommend to install it today.

Comment (d7415) [reply README.md]: Agreed, it just feels weird saying "We recommend using Docker, but first, here's how to do it a different way" :)

Comment (tinohager) [reply README.md]: I turned it

File: README.md [hunk 2]
```diff
@@ -35,14 +35,14 @@ Through [contributions, donations, and sponsorship](https://github.com/sponsors/
 
 There are three deployment options available. The simplest is **Docker deployment**, as all dependencies are already predefined and configured.
 
-### File-based deployment
-
-Copy the extracted Zip file from https://github.com/LycheeOrg/Lychee/releases to your webserver
-
 ### Docker deployment
 
 Alternatively, an official Docker image can be found at [LycheeOrg/Lychee-Docker](https://github.com/LycheeOrg/Lychee-Docker) or on Docker Hub as [lycheeorg/lychee](https://hub.docker.com/r/lycheeorg/lychee).
```
Comment (d7415) [inline README.md]:
```suggestion
An official Docker image can be found at [LycheeOrg/Lychee-Docker](https://github.com/LycheeOrg/Lychee-Docker) or on Docker Hub as [lycheeorg/lychee](https://hub.docker.com/r/lycheeorg/lychee).
```

Minor tweak with the reordering :)

**Review by d7415 — APPROVED** [review]

**Pull request merged**